functions.php

```diff
@@ -349,6 +349,24 @@ function ($urls, $post_id) {
     2
 );
 
+// Add filters to the News & Stories listing page.
+// Right now only "category" is available.
+add_action(
+    'pre_get_posts',
+    function ($query): void {
+        if (!$query->is_main_query() || is_admin() || !is_home()) {
+            return;
+        }
+        $category_slug = isset($_GET['category']) ? $_GET['category'] : '';
+        if (!empty($category_slug)) {
+            $category = get_category_by_slug($category_slug);
```

Comment:
If `get_category_by_slug($category_slug)` doesn’t find a matching category, it will return false, which will cause an error when you try to access `$category->term_id`.

Reply:
Updated